tsdb/head_test.go

```diff
@@ -6348,10 +6476,21 @@ func TestHeadAppender_AppendCTZeroSample(t *testing.T) {
 			a := h.Appender(context.Background())
 			lbls := labels.FromStrings("foo", "bar")
 			for _, sample := range tc.appendableSamples {
-				_, err := a.AppendCTZeroSample(0, lbls, sample.ts, sample.ct)
-				require.NoError(t, err)
-				_, err = a.Append(0, lbls, sample.ts, sample.val)
-				require.NoError(t, err)
+				// Append float if it's a float test case
+				if sample.fSample != 0 {
+					_, err := a.AppendCTZeroSample(0, lbls, sample.ts, sample.ct)
+					require.NoError(t, err)
+					_, err = a.Append(0, lbls, sample.ts, sample.fSample)
+					require.NoError(t, err)
+				}
+
+				// Append histograms if it's a histogram test case
+				if sample.h != nil || sample.fh != nil {
+					ref, err := a.AppendHistogramCTZeroSample(0, lbls, sample.ts, sample.ct, sample.h, sample.fh)
+					require.NoError(t, err)
+					_, err = a.AppendHistogram(ref, lbls, sample.ts, sample.h, sample.fh)
+					require.NoError(t, err)
+				}
 			}
 			require.NoError(t, a.Commit())
```

Comment:
nit: it's kind of weird that we don't test with multiple Commits() , thus the path to return out of order is never exercised. As far as I can tell that was the case for floats anyway, so maybe another PR?

Reply:
Interesting. Yeah, we haven't tested OOO well enough, that was a good call out. If that's ok I'd be happy to extend the tests here to cover OOO in another PR